e2e/pod_status.go

```diff
@@ -49,6 +50,45 @@ func isPodGone(cs *kubernetes.Clientset, podName, namespace string) wait.Conditi
 	}
 }
 
+func isStatefulSetRunning(cs *kubernetes.Clientset, serviceName string, namespace string, expectedReplicas int) wait.ConditionFunc {
+	return func() (bool, error) {
+		statefulSet, err := cs.AppsV1().StatefulSets(namespace).Get(context.Background(), serviceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		return statefulSet.Status.ReadyReplicas == int32(expectedReplicas), nil
+	}
+}
+
+func isStatefulSetGone(cs *kubernetes.Clientset, serviceName string, namespace string, labelSelector string) wait.ConditionFunc {
+	return func() (done bool, err error) {
+		statefulSet, err := cs.AppsV1().StatefulSets(namespace).Get(context.Background(), serviceName, metav1.GetOptions{})
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return false, fmt.Errorf("something weird happened with the stateful set whose status is: [%s]. Errors: %w", statefulSet.Status.String(), err)
+		}
+
+		associatedPods, err := cs.CoreV1().Pods(namespace).List(context.TODO(), selectViaLabels(labelSelector))
+		if err != nil {
+			return false, err
+		}
+
+		return isStatefulSetEmpty(statefulSet) && areAssociatedPodsGone(associatedPods), nil
```

Comment:
Garbage collection (foreground cascading deletion) won't work for cleaning up pods, I guess?

Reply:
It does, I just haven't used it yet.